htdocs/core/class/html.formfile.class.php

```diff
@@ -1320,5 +1328,34 @@ public function listOfLinks($object, $permtodelete=1, $action=null, $selected=nu
         return $nboflinks;
     }
 
+    /**
+     * Show detail icon with link for preview
+     * @param array     $file   File
+     * @param string    $modulepart     propal, facture, facture_fourn, ...
+     * @param string    $relativepath   Relative path of docs
+     * @return string   $out            Output string with HTML
+     */
+    function showPreview($file, $modulepart, $relativepath){
+        global $langs;
+
+        $out='';
+
+        $mime_preview = array('jpeg', 'png', 'gif', 'tiff', 'pdf', 'plain', 'css');
+        $num_mime = array_search(dol_mimetype($file['name'], '', 1), $mime_preview);
+
+        if( $num_mime!== false){
+            $out.= ' <a href="javascript:document_preview(\''.DOL_URL_ROOT.'/document.php?modulepart='.$modulepart.'&amp;attachment=0&amp;file='.urlencode($relativepath).'\',\''.dol_mimetype($file['name']).'\', \''.$file['name'].'\'';
+            if($num_mime<=3){
+                $size = getimagesize($file[fullname]);
```

Comment:
The image is 4k, and popup is large 90% of user screen.
I not use $_SESSION["dol_screenwidth"] and $_SESSION["dol_screenheight"] because the user can resize the window, i check in javascript the current size ( $(window).width() and  $(window).height()  ).

![4k](https://cloud.githubusercontent.com/assets/6471493/17296386/57c29d48-5801-11e6-95a2-96e6bdb0bddf.png)

I send the size because the image is small then it's not necessary 90%... for example

![auriculares](https://cloud.githubusercontent.com/assets/6471493/17296873/85076be2-5803-11e6-85cc-335ed9705757.png)